File: scripts/get_certificate_arn.rb
require 'aws-sdk'
require 'optparse'

option = ARGV.getopts(nil, 'domain:', 'region:')
           .inject({}) { |hash, (k, v)| hash[k.to_sym] = v; hash }

acm = Aws::ACM::Client.new(region: option[:region])
puts acm.list_certificates
       .certificate_summary_list
       .select { |c| c.domain_name == option[:domain] }
       .map(&:certificate_arn)
       .first

File: scripts/get_certificate_arn.js
const AWS = require('aws-sdk');
const ArgumentParser = require('argparse').ArgumentParser;

AWS.config.region = process.env.AWS_DEFAULT_REGION;

function parseArguments() {
  const parser = new ArgumentParser({
    description: 'This script find an ACM arn by domain name',
  });
  parser.addArgument(['-d', '--domain'], {
    help: 'domain name of certificate to search',
    required: true,
  });
  return parser.parseArgs();
}

const args = parseArguments();
const ACM = new AWS.ACM();

ACM.listCertificates((err, data) => {
  const arn = data.CertificateSummaryList
    .filter(s => s.DomainName === args.domain)
    .map(s => s.CertificateArn);
  console.log(arn[0]);
});

